src/lib/message/L1ToL2MessageGasEstimator.ts

```diff
@@ -253,4 +261,93 @@ export class L1ToL2MessageGasEstimator {
       deposit,
     }
   }
+
+  /**
+   * Transactions that make an L1->L2 message need to estimate L2 gas parameters
+   * This function does that, and populates those parameters into a transaction request
+   * @param dataFunc
+   * @param l1Provider
+   * @param gasOverrides
+   * @returns
+   */
+  public async populateL1ToL2FunctionParams(
+    /**
+     * Function that will internally make an L1->L2 transaction
+     * Will initially be called with dummy values to trigger a special revert containing
+     * the real params. Then called again with the real params to form the final data to be submitted
+     */
+    dataFunc: (
+      params: OmitTyped<L1ToL2MessageGasParams, 'deposit'>
+    ) => L1ToL2TransactionRequest['core'],
+    l1Provider: Provider,
+    gasOverrides?: GasOverrides
+  ) {
+    // get function data that should trigger a retryable data error
+    const {
+      data: nullData,
+      to,
+      value,
+      from,
+    } = dataFunc({
+      gasLimit: RetryableDataTools.ErrorTriggeringParams.gasLimit,
+      maxFeePerGas: RetryableDataTools.ErrorTriggeringParams.maxFeePerGas,
+      maxSubmissionCost: BigNumber.from(1),
+    })
+
+    let retryable: RetryableData | null
+    try {
+      // get retryable data from the null call
+      const res = await l1Provider.call({
+        to: to,
+        data: nullData,
+        value: value,
+        from: from,
+      })
+      retryable = RetryableDataTools.tryParseError(res)
+      if (!isDefined(retryable)) {
+        throw new ArbSdkError(`No retryable data found in error: ${res}`)
+      }
+    } catch (err) {
+      retryable = RetryableDataTools.tryParseError(err as Error)
```

Comment:
would be good to add a comment explaining https://github.com/OffchainLabs/arbitrum-sdk/pull/149#discussion_r952677274